Engine/DataFeeds/Enumerators/SynchronizingEnumerator.cs

```diff
@@ -200,7 +199,7 @@ private static void ListInsert(SynchronizedEnumerator[] list, int index, Synchro
         private static IEnumerator<BaseData> GetBruteForceMethod(IEnumerator<BaseData>[] enumerators)
         {
             var ticks = DateTime.MaxValue.Ticks;
-            var collection = new ConcurrentDictionary<IEnumerator<BaseData>, int>();
+            var collection = new Dictionary<IEnumerator<BaseData>, int>();
```

Comment:
Why not just use a `HashSet<T>`-- it's faster than a dictionary and it doesn't look like you're they the key/value -- just the key here.

Reply:
Good idea 👍